### PR TITLE
refactor device app to separate common functionality.  add service ap…

### PR DESCRIPTION
**Pull request by BertKleewein**
…p.  remove unnecessary objects

### CODE REVIEW

File: scripts/fetch-secrets.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    exit 33
+fi
+
+THIEF_KEYVAULT_NAME="thief-kv"
+
+function get-secret {
+    bash_name=$1
+    kv_name=$2
+    echo "Fetching ${bash_name}"
+    value=$(az keyvault secret show --vault-name ${THIEF_KEYVAULT_NAME} --name ${kv_name} | jq -r ".value")
```
Comment (YoDaMa) [inline scripts/fetch-secrets.sh:16]: for my sanity: how is this being run? I assume locally, so you're already logged into your azure account via the CLI?

File: python/device/device.py [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+        self.last_heartbeat = time.time()
+        self.shutdown_event = threading.Event()
+
+        self.start_reaper()
+
+    def update_initial_properties(self):
+        """
+        Update reported properties at the start of a run
+        """
+        # toto: update these values
```
Comment (YoDaMa) [inline python/device/device.py:95]: is this still todo?

File: python/device/device.py [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+        """
+
+        def send_single_d2c_message():
+            message_id = str(uuid.uuid4())
+            data = make_new_d2c_payload(message_id)
+            latency = MeasureLatency()
+
+            try:
+                self.metrics.d2c.inflight.increment()
+                with latency:
```
Comment (YoDaMa) [inline python/device/device.py:117]: this is an interesting design for latency...

File: python/device/device.py [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+                obj = json.loads(msg.data.decode())
+                if obj.get("thiefHeartbeat"):
+                    logger.info("heartbeat received")
+                    self.last_heartbeat = time.time()
+                elif obj.get("thiefPingbackResponse"):
+                    list = obj.get("messageIds")
+                    if list:
+                        with self.d2c_set_lock:
+                            for message_id in list:
+                                self.d2c_confirmed.add(message_id)
```
Comment (YoDaMa) [inline python/device/device.py:207]: what is the difference between d2c_confirmed and d2c_unconfirmed, i.e. what do these lists mean?

File: python/containers/fetch-service-secrets.sh [hunk 5]
```diff
@@ -1,21 +1,9 @@
-THIEF_KEYVAULT_NAME="thief-kv"
+# Copyright (c) Microsoft. All rights reserved.
```
Comment (YoDaMa) [inline python/containers/fetch-service-secrets.sh:1]: not consistent with the copyright

File: python/containers/fetch-functions.sh [hunk 6]
```diff
@@ -0,0 +1,19 @@
+export THIEF_KEYVAULT_NAME="thief-kv"
+
+echo "Getting token"
+while [ "$token" == "" ]; do 
+    response=$(curl --silent 'http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https%3A%2F%2Fvault.azure.net' -H Metadata:true)
```
Comment (drwill-ms) [inline python/containers/fetch-functions.sh:5]: hard coded IP address?

File: python/device/device.py (hunk 2 above)
Comment (drwill-ms) [reply python/device/device.py:95]: No it is a [toto](https://www.youtube.com/watch?v=FTQbiNvZqaY), which is totally different!

File: python/service/service.py [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+        """
+        if self.shutdown_event.isSet():
+            logger.info("shutdown: event is already set.  ignorning")
+        else:
+            if error:
+                self.metrics.run_state = longhaul.FAILED
+                logger.error("shutdown: triggering error shutdown", exc_info=error)
+            else:
+                self.metrics.run_state = longhaul.COMPLETE
+                logger.info("shutdown: trigering clean shutdown")
```
Comment (drwill-ms) [inline python/service/service.py:192]: misspelled triggering

File: scripts/fetch-secrets.sh (hunk 1 above)
Comment (drwill-ms) [reply scripts/fetch-secrets.sh:16]: Worth a comment about how to set up an environment so this runs.

**Review by drwill-ms — APPROVED** [review]

File: python/device/device.py (hunk 2 above)
Comment (BertKleewein) [reply python/device/device.py:95]: Yes, it's still a toto.  That dog is going to come through one of these days.  It's also a todo

---
In reply to: [491170812](https://github.com/Azure/iot-sdk-longhaul/pull/9#discussion_r491170812) [](ancestors = 491170812)

File: scripts/fetch-secrets.sh (hunk 1 above)
Comment (BertKleewein) [reply scripts/fetch-secrets.sh:16]: Yes, this gets called manually.  Docs coming.

---
In reply to: [491169089](https://github.com/Azure/iot-sdk-longhaul/pull/9#discussion_r491169089) [](ancestors = 491169089)

Comment (BertKleewein) [reply scripts/fetch-secrets.sh:16]: doc is coming in my next review.

---
In reply to: [492340762](https://github.com/Azure/iot-sdk-longhaul/pull/9#discussion_r492340762) [](ancestors = 492340762)

File: python/containers/fetch-functions.sh (hunk 6 above)
Comment (BertKleewein) [reply python/containers/fetch-functions.sh:5]: Yes.  169.254.169.254 is a well-known IP used to get identity data from Azure.  see [https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service) 

---
In reply to: [492338139](https://github.com/Azure/iot-sdk-longhaul/pull/9#discussion_r492338139) [](ancestors = 492338139)

Comment (drwill-ms) [reply python/containers/fetch-functions.sh:5]: Worth a code comment, or better yet an aptly named variable then

File: python/device/device.py (hunk 4 above)
Comment (BertKleewein) [reply python/device/device.py:207]: I'm going to rewrite this section of code.

**Pull request merged**